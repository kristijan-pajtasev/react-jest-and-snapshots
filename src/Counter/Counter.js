import React, {useState} from 'react';

const Counter = () => {
    const [counterValue, setCounterValue] = useState(0);

    const incrementValue = () => setCounterValue(counterValue + 1);

    return (
        <div>
            {counterValue}
            <button onClick={incrementValue}>Increment</button>
        </div>
    );
};

export default Counter;