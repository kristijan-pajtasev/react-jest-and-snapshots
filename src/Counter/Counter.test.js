import React from 'react';
import Counter from './Counter';
import ShallowRenderer from 'react-test-renderer/shallow';

describe("Counter stateless component ", () => {

    it('renders without crashing', () => {

        const renderer = new ShallowRenderer();
        const tree = renderer
            .render(<Counter/>);
        expect(tree).toMatchSnapshot();
    });
});
