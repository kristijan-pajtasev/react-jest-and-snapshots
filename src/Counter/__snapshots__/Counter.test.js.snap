// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter stateless component  renders without crashing 1`] = `
<div>
  0
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
